fixql-core/src/main/scala/com/iterable/graphql/FromGraphQLJava.scala

```diff
@@ -0,0 +1,202 @@
+          //val parentType: GraphQLFieldsContainer = getFieldsContainer(executionInfo)
+          //val argumentValues: util.Map[String, AnyRef] = valuesResolver.getArgumentValues(codeRegistry, fieldDef.getArguments, field.getArguments, executionContext.getVariables)
+          //val fieldDef: GraphQLFieldDefinition = executionInfo.getFieldDefinition
+          //val fieldType: GraphQLOutputType = fieldDef.getType
+          val argumentValues = valuesResolver.getArgumentValues(fieldDef.getArguments, mergedField.getArguments, context.getVariables)
+          val selectionSet = DataFetchingFieldSelectionSetImpl.newCollector(context, unwrappedType, mergedField)
+          mkField(mergedField.getName, argumentValues, selectionSet)
+        }.toSeq
+      Query[Field.Fixed](topLevelFields)
+    }
+  }
+
+  private def toJavaValues(value: Json): AnyRef = {
+    value.foldWith(new Json.Folder[AnyRef] {
+      override def onNull = null
+      override def onBoolean(value: Boolean) = value: java.lang.Boolean
+      override def onNumber(value: JsonNumber) = value.toBigDecimal
+      override def onString(value: String) = value
+      override def onArray(value: Vector[Json]) = value.map(toJavaValues)
+      override def onObject(value: JsonObject) = value.toMap.mapValues(toJavaValues)
+    })
+  }
+
+  import qq.droste.syntax.fix._
+  private def mkField(fieldName: String, arguments: java.util.Map[String, AnyRef], fss: DataFetchingFieldSelectionSet): Field.Fixed = {
+    // getFields contains flattened fields from all children, but we only want the immediate children
+    val childFields = fss.getFields.asScala.filterNot(_.getQualifiedName.contains("/"))
+    val args = PlayJson.obj(arguments.asScala.map { case (key, value) =>
+      key -> (fromJavaValue(value): play.api.libs.json.Json.JsValueWrapper)
+    }.toSeq: _*)
+    Field[Field.Fixed](
+      fieldName,
+      args,
+      childFields.map { selectedField =>
+        val subfss = selectedField.getSelectionSet
+        mkField(selectedField.getName, selectedField.getArguments, subfss)
+      }.toSeq,
+    ).fix
+  }
+
+  private def fromJavaValue(value: Any): JsValue = {
+    value match {
+      case x if x == null => JsNull
+      case f: Float => JsNumber(BigDecimal.decimal(f))
+      case s: String => JsString(s)
+      case n: Int => JsNumber(n)
+      case n: Long => JsNumber(n)
+      case f: Boolean => JsBoolean(f)
+      case rg: Array[_] => JsArray(rg.map(fromJavaValue))
+    }
+  }
+
+  /**
+    * Create a GraphQL Java GraphQLSchema from a GraphQL SDL String.
+    */
+  def parseSchema(schemaStr: String): GraphQLSchema = {
+    val schemaParser = new SchemaParser
+    val schemaGenerator = new SchemaGenerator
+
+    val typeRegistry = schemaParser.parse(schemaStr)
+    val wiring = RuntimeWiring.newRuntimeWiring().wiringFactory(new EchoingWiringFactory).build()
+    val graphQLSchema = schemaGenerator.makeExecutableSchema(typeRegistry, wiring)
```

Comment:
nit: don't need the `val`

Reply:
i like keeping these just for clarity and ease of changing the code in the future